src/OrganisationRegistry.Api/Security/SecurityService.cs

```diff
@@ -228,16 +212,66 @@ public bool CanUseLabelType(IUser user, Guid labelTypeId)
             return true;
         }
 
+        private static bool HasPermissionsForOrganisation(SecurityInformation securityInfo, Guid? organisationId)
+        {
+            if (!organisationId.HasValue)
+                return false;
+
+            return
+                securityInfo.Roles.Contains(Role.OrganisatieBeheerder) &&
+                securityInfo.OrganisationIds.Contains(organisationId.Value);
+        }
+
+        private static bool HasPermissionsForBody(SecurityInformation securityInfo, Guid? bodyId)
+        {
+            if (!bodyId.HasValue)
+                return false;
+
+            return
+                securityInfo.Roles.Contains(Role.OrganisatieBeheerder) &&
+                securityInfo.BodyIds.Contains(bodyId.Value);
+        }
+
+        private async Task<OrganisationSecurityInformation> GetOrganisationSecurityInformation(
+            ClaimsPrincipal user,
+            bool isAuthenticated,
+            string subject)
+        {
+            var maybeCachedSecurityInfo = isAuthenticated ? _cache.Get(subject) : null;
```

Comment:
I think it would improve readability if you started with:
`if (!isAuthenticated)
    return new OrganisationSecurityInformation();`

then you can remove all isAuthenticated checks from the rest of the method